keras/backend/mxnet_backend.py

```diff
@@ -2690,55 +2654,394 @@ def rnn(step_function, inputs, initial_states,
     global uses_learning_phase
     uses_learning_phase = False
 
-    states = initial_states
-    outputs = []
-    prev_output = None
-
-    if mask is not None:
-        if not states:
-            raise ValueError('MXNet Backend: Initial states is not provided when masking is '
-                             'enabled.')
-        if mask.dtype != dtype:
-            mask = cast(mask, dtype)
-        # Split the mask across time dimension and generate the list of masks
-        # with shape `(samples, 1)` (no time dimension)
-        masks = list(mx.sym.split(mask.symbol, axis=1,
-                                  squeeze_axis=1, num_outputs=dshape[1]))
-        # Reverse the mask sequence
+    if unroll:
+        # Split the inputs across time dimension and generate the list of inputs
+        # with shape `(samples, ...)` (no time dimension)
+        inputs = list(mx.sym.split(inputs.symbol, axis=1,
+                                   squeeze_axis=1, num_outputs=dshape[1]))
+
+        # Reverse the input sequence
         if go_backwards:
-            masks.reverse()
+            inputs.reverse()
+
+        states = initial_states
+        outputs = []
+        prev_output = None
+
+        if mask is not None:
+            if not states:
+                raise ValueError('MXNet Backend: Initial states is not provided when masking is '
+                                 'enabled.')
+            if mask.dtype != dtype:
+                mask = cast(mask, dtype)
+            # Split the mask across time dimension and generate the list of masks
+            # with shape `(samples, 1)` (no time dimension)
+            masks = list(mx.sym.split(mask.symbol, axis=1,
+                                      squeeze_axis=1, num_outputs=dshape[1]))
+            # Reverse the mask sequence
```

Comment:
remove comment

Reply:
This is from original rnn implementation with unrolling, let's keep it